web/src/features/fbaCalculator/components/grassCureCell.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { render, screen } from '@testing-library/react'
+import { FBAInputRow } from 'features/fbaCalculator/components/FBAInputGrid'
+import GrassCureCell, {
+  GrassCureCellProps
+} from 'features/fbaCalculator/components/GrassCureCell'
+import { FBAFuelType, FuelTypes } from 'features/fbaCalculator/fuelTypes'
+import { isNull } from 'lodash'
+import React from 'react'
+describe('GrassCureCell', () => {
+  const buildProps = (inputRow: FBAInputRow, value?: number): GrassCureCellProps => ({
+    fbaInputGridProps: {
+      inputRows: [inputRow],
+      updateRow: () => {
+        /** no op */
+      }
+    },
+    classNameMap: { grassCure: '' },
+    value,
+    rowId: 0
+  })
+  describe('grass cure failure states', () => {
+    const buildInputRow = (fuelType: 'o1a' | 'o1b') => ({
+      id: 0,
+      weatherStation: undefined,
+      fuelType: fuelType,
+      grassCure: undefined,
+      windSpeed: undefined
+    })
+    it('should return field in error state o1a without percentage set', () => {
+      const zero1ARow = buildInputRow('o1a')
+      const props = buildProps(zero1ARow)
+      const { container } = render(<GrassCureCell {...props} />)
+      expect(container.firstChild?.firstChild).toHaveClass('Mui-error')
+    })
+    it('should return field in error state o1b without percentage set', () => {
+      const zero1BRow = buildInputRow('o1b')
+      const props = buildProps(zero1BRow)
+      const { container } = render(<GrassCureCell {...props} />)
+      expect(container.firstChild?.firstChild).toHaveClass('Mui-error')
+    })
+  })
+  describe('grass cure successful states', () => {
+    const buildInputRow = (fuelType: FBAFuelType | null): FBAInputRow => {
+      if (isNull(fuelType)) {
+        fail('Got null fuel type')
+      }
+      return {
+        id: 0,
+        weatherStation: undefined,
+        fuelType: fuelType.name,
+        grassCure: 1,
+        windSpeed: undefined
+      }
+    }
+    it('should return field without error state when o1a when percentage set', () => {
+      const zero1ARow = buildInputRow(FuelTypes.lookup('o1a'))
+      const props = buildProps(zero1ARow, 1)
+      render(<GrassCureCell {...props} />)
+      const renderedGrassCureInputField = screen.getByTestId(`grassCureInput-0`)
+        .firstChild?.firstChild
+      expect(renderedGrassCureInputField).toHaveValue(1)
+      expect(renderedGrassCureInputField).not.toHaveClass('Mui-error')
+    })
+    it('should return field in error state o1b without percentage set', () => {
+      const zero1BRow = buildInputRow(FuelTypes.lookup('o1b'))
+      const props = buildProps(zero1BRow, 1)
+      render(<GrassCureCell {...props} />)
+      const renderedGrassCureInputField = screen.getByTestId(`grassCureInput-0`)
+        .firstChild?.firstChild
+      expect(renderedGrassCureInputField).toHaveValue(1)
+      expect(renderedGrassCureInputField).not.toHaveClass('Mui-error')
+    })
```

Comment:
FYI - if you're using the stepper buttons in the grass cure cell, the value maxes out at 100, but if you type in the value on a keyboard, you can still set values > 100 and the HFI, etc. calculations will increase accordingly (e.g., grass cure of 100 has HFI 4500; grass cure of 150 has HFI 5500).

Is this an opportunity for another unit test? A bug to be fixed in this ticket? Or in a separate ticket? Don't know. Just logging this while I still remember.

Reply:
I think I can fix that issue in this PR, thanks!

Reply:
I left a message in the General channel on Friday because I thought that capping the grass cure at a max of 100 might have been an assumption that we made, rather than knowing for sure, and I was feeling somewhat hesitant because it is theoretically possible for RH to go above 100. EK replied saying he wasn't sure whether grass cure can go above 100. Didn't get a reply from BB or DH - it miiiiight be worth confirming before you go to the trouble of actually doing the work.

Reply:
Ok cool thanks for letting me know, I wouldn't have thought of that. I'll hold off on this change until it's confirmed then.

Reply:
Limited grass cure input to 100 both via stepper and keyboard.